src/foam/dao/RestDAO.js

```diff
@@ -87,6 +88,17 @@ foam.CLASS({
        * Each key's value is network-foam-jsonified.
        */
       var query = [];
+
+      // Proxy non-serializable sinks (with either "sink" or an ArraySink).
+      // When "sink" is serializable, make "proxySink" falsey as a signal to:
+      // (1) Send the serializable sink (" ! proxySink" below);
+      // (2) Signal to "onSelectResponse" that sink in payload should be passed
+      //     back to caller.
+      var proxySink = this.Serializable.isInstance(sink) ? null :
```

Comment:
Isn't this design simpler and more direct:

 var networkSink = this.Serializable.isInstance(sink) && sink;
      if ( networkSink )
        query.push('sink=' + encodeURIComponent(this.jsonify_(networkSink)));

...

     .then(this.onSelectResponse);


 function onSelectResponse(localSink, payload) {
      var wasSerializable =  this.Serializable.isInstance(localSink);
      var remoteSink = foam.json.parse(payload);

      // If not proxying a local unserializable sink, just return the remote.
      if ( wasSerializable ) return remoteSink; // Or could have been passed as a parameter

      var array = remoteSink.a;
      if ( ! array )
        throw new Error('Expected ArraySink from REST endpoint when proxying local sink');

      if ( localSink.put ) {
        for ( var i = 0; i < array.length; i++ ) {
          localSink.put(array[i]);
        }
      }
      if ( localSink.eof ) localSink.eof();

      return localSink;
    },